sdk/core-rest/core-client/src/getClient.ts

```diff
@@ -0,0 +1,140 @@
+    trace: (options?: RequestParameters) => Promise<PathUncheckedResponse>;
+  };
+}
+
+/**
+ * Creates a client with a default pipeline
+ * @param baseUrl Base endpoint for the client
+ * @param options Client options
+ */
+export function getClient(baseUrl: string, options?: PipelineOptions): Client;
+/**
+ * Creates a client with a default pipeline
+ * @param baseUrl Base endpoint for the client
+ * @param credentials Credentials to authenticate the requests
+ * @param options Client options
+ */
+export function getClient(
+  baseUrl: string,
+  credentials?: TokenCredential | KeyCredential,
+  options?: PipelineOptions
+): Client;
+export function getClient(
+  baseUrl: string,
+  credentialsOrPipelineOptions?: (TokenCredential | KeyCredential) | PipelineOptions,
+  opts: PipelineOptions = {}
+): Client {
+  let credentials: TokenCredential | KeyCredential | undefined;
+  let options = opts;
+
+  if (credentialsOrPipelineOptions) {
+    if (isCredential(credentialsOrPipelineOptions)) {
+      credentials = credentialsOrPipelineOptions;
+      options = opts;
+    } else {
+      options = credentialsOrPipelineOptions || {};
+    }
+  }
+
+  const pipeline = createDefaultPipeline(baseUrl, credentials, options);
+  pipeline.removePolicy({ name: "exponentialRetryPolicy" });
+  const client = (path: string, ...args: Array<any>) => {
+    return {
+      get: (options: RequestParameters = {}): Promise<HttpResponse> => {
+        const url = buildRequestUrl(baseUrl, path, args, options);
+        return sendRequest("GET", url, pipeline, options);
+      },
+      post: (options: RequestParameters = {}): Promise<HttpResponse> => {
+        const url = buildRequestUrl(baseUrl, path, args, options);
+        return sendRequest("POST", url, pipeline, options);
+      },
+      put: (options: RequestParameters = {}): Promise<HttpResponse> => {
+        const url = buildRequestUrl(baseUrl, path, args, options);
+        return sendRequest("PUT", url, pipeline, options);
+      },
+      patch: (options: RequestParameters = {}): Promise<HttpResponse> => {
+        const url = buildRequestUrl(baseUrl, path, args, options);
+        return sendRequest("PATCH", url, pipeline, options);
+      },
+      delete: (options: RequestParameters = {}): Promise<HttpResponse> => {
+        const url = buildRequestUrl(baseUrl, path, args, options);
+        return sendRequest("DELETE", url, pipeline, options);
+      },
+      head: (options: RequestParameters = {}): Promise<HttpResponse> => {
+        const url = buildRequestUrl(baseUrl, path, args, options);
+        return sendRequest("HEAD", url, pipeline, options);
+      },
+      options: (options: RequestParameters = {}): Promise<HttpResponse> => {
+        const url = buildRequestUrl(baseUrl, path, args, options);
+        return sendRequest("OPTIONS", url, pipeline, options);
+      },
+      trace: (options: RequestParameters = {}): Promise<HttpResponse> => {
```

Comment:
you know, I've never seen a service use `TRACE` before. Do we actually use it somewhere?

Reply:
I haven't seen it in use either. Just added it for consistency with core-rest-pipeline HttpMethod.

Reply:
Yeah, I think we might as well include it for completeness, was more just amused by it

Reply:
I searched all our public swaggers and didn't find a single instance of trace. I think it's fine to not support `trace` for now.

I can't find conclusive evidence of `options` in the swagger but considering it's part of the CORS handshake I know services support it. Fine keeping that one.